src/runtime-pmi.c

```diff
@@ -266,7 +274,7 @@ shmem_runtime_get(int pe, char *key, void *value, size_t valuelen)
         HASH_FIND_STR(singleton_kvs, kvs_key, e);
         if (e == NULL)
             return 3;
-        kvs_value = e->val;
+        *kvs_value = *(e->val);
```

Comment:
I think what you really want here is:
```
strncpy(kvs_value, e->val, max_val_len);
```

Reply:
Otherwise, you are just copying a single character. :)

Reply:
Thanks @jdinan!